src/controller/custom_controller.py

```
@@ -29,3 +29,40 @@ async def paste_from_clipboard(browser: BrowserContext):
             await page.keyboard.type(text)
 
             return ActionResult(extracted_content=text)
+
+        @self.registry.action('Upload file to element',requires_browser=True)
+        async def upload_file(index: int, browser: BrowserContext):
```

Ruff (0.8.2)

Missing return type annotation for private function `upload_file`

(ANN202)